### DISCUSSION TITLE
Is there any way of optimizing assets GIF file size

### DISCUSSION THREAD

**Discussion by Pejosonic** [directus/directus#17001] [Archive]
Hi communitty!
I have created some assets transformation rules for fetching thumbnails of uploaded images.
Some of these images are GIFs files and i see that they are being bypassed from the resizing function or convert to webp format.

Is there any automated way to reduce the size of GIF files?

When reading the sharp documentations looks like its possible. But by reading the Directus source looks like it's implementing a restricted subset of the sharp API. In Directus docs says that its available the full sharp API but just managed to get working a few methods.

I think that maybe someone have stepped into the same problem.
Thank you!

**Comment by paescuj** [comment]
I think this is currently not supported because the chances are high that the transformation of GIF files must be treated specially, e.g. setting the `{ animated: true }` option and requiring`libvips` to be compiled with [`giflb`](https://github.com/lovell/sharp/issues/2395#issuecomment-703501431) (don't know whether this is the case by default).

**Comment by Pejosonic** [comment]
Thank you for your answer @paescuj
I'm trying another approach to handle this situation and now i'm stuck in another step. I decider to convert all gif images to webp in order to get them optimized.
Now the problem is that when directus optimize an animated webp it's loosing the animation.
Is there a way to set `{ animated: true }` from the directus fron end transformation presets?

Thank you!

**Reply by paescuj** [reply]
You're welcome!

I'm afraid it's currently not possible to set the `animated` option, as this seems to be a global option (see this [example](https://sharp.pixelplumbing.com/api-output#examples-8)) rather than the [image operations](https://sharp.pixelplumbing.com/api-operation) which is what Directus currently exposes via [advanced transformations](https://docs.directus.io/reference/files.html#advanced-transformations).

I'm going to bring this up in #17076.

**Comment by Pejosonic** [comment]
Thank you for your reply Pascal.
I feel like I'm trapped in a bottleneck by trying to allow some users to
upload animated logos for their profiles.
If they use GIF files, they cannot be resized by directus.
If they use webp the loose animation on resizing and also are displayed
static in directus admin (less important for me).

I found a way to go around this problem by combining external image
processors so this problem isn't being a problem anymore.
Anyways I would like to learn how it would be the optimal way for handling
this scenario by using only Directus.

**Reply by paescuj** [reply]
You're welcome!

I'm glad that you have found an acceptable solution for the moment.
However, I understand that this is not entirely satisfactory. Therefore, I've amended the corresponding issue (#17076) accordingly and I'm pretty sure that we can come up with an appropriate solution here in the future.
